File: leetcode_ly/c++/Array/ContainerWithMostWater.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int begin = 0;
        int end = height.size() - 1;
        int area = 0;
        int max = INT_MIN;
        
        while(begin <= end){
            while(height[begin] <= height[end] && begin <= end){
                area = (end - begin) * height[begin];
                max = (max > area) ? max : area;
                begin++;
            }
            while(height[begin] > height[end] && begin <= end){
                area = (end - begin) * height[end];
                max = (max > area) ? max : area;
                end--;
            }
        }
        return (max == INT_MIN) ? 0 : max;
    }
};

File: leetcode_ly/java/Array/ContainerWithMostWater.java
public class Solution {
    public int maxArea(int[] height) {
        int i = 0;
        int j = height.length - 1;
        int res = Integer.MIN_VALUE;
        while(i < j){
            res = Math.max(res,Math.min(height[i],height[j]) * (j - i));
            if(height[i] < height[j]) i++;
            else j--;
        }
        return res;
    }
}